src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -189,6 +189,90 @@ ShardManager::AsyncResult< ShardInfo > HSHomeObject::_seal_shard(ShardInfo const
     return req->result();
 }
 
+// move seal_shard to pre_commit can not fundamentally solve the conflict between seal_shard and put_blob, since
+// put_blob handler will only check the shard state at the very beginning and will not check again before proposing to
+// raft, so we need a callback to check whether we can handle this request before appending log, which is previous to
+// pre_commit.
```

Comment:
prior to the pre_commit of put_blob

Reply:
thanks for pointing out this misusing , will be careful!